firestore/src/Connector.gs

```diff
@@ -0,0 +1,288 @@
+               you must supply a set of fields in the collection you want to use. \
+               Use the following format: <field_name1>:<type1>, where type is one of STRING, \
+               NUMBER, BOOLEAN, or TIMESTAMP'
+      },
+      {
+        name: 'schema',
+        type: 'TEXTAREA',
+        displayName: 'Firestore "Schema"',
+        placeholder: 'field1:STRING\nfield2:STRING\nfield3:NUMBER\nfield4:BOOLEAN'
+      },
+      {
+        name: 'numResults',
+        type: 'SELECT_SINGLE',
+        displayName: 'Max Documents',
+        helpText: 'Specifies the number of documents to read on each request. Note that each page \
+                   load will fetch *all* the documents, so setting this too high may be expensive. \
+                   Consider pre-processing in Firestore if there are many documents. Defaults to 1000.',
+        options: [
+          {label: '100', value: '100'},
+          {label: '1000', value: '1000'},
+          {label: '10000', value: '10000'},
+          {label: '100000', value: '100000'},
+        ]        
+      },
+      {
+        name: 'authEmail',
+        type: 'TEXTINPUT',
+        displayName: 'Auth Email (optional)',
+        helpText: 'Optional field for specifying authentication email for Firestore.'
+      },
+      {
+        name: 'authKey',
+        type: 'TEXTINPUT',
+        displayName: 'Auth Private Key (optional)',
+        helpText: 'Optional field for specifying authentication key for Firestore.'
+      }
+    ]
+  };
+}
+
+
+/** 
+ * Predefined Data Studio function for generating schema from user input. 
+ * 
+ * @param {Object} request Schema request parameters.
+ * @return {Object} Schema for the given request.
+ */
+Connector.prototype.getSchema = function(request) {
+  // Convert the user input to a Data Studio schema
+  const configCsv = request.configParams.schema;
+  var schema = this.parseSchemaCsv(configCsv);
+  
+  // Add predefined fields
+  schema.push({name: 'name', label: 'id', dataType: 'STRING'});
+  schema.push({name: 'createTime', label: 'created', dataType: 'STRING', semantics: this.TIMESTAMP_SEMANTICS});
+  schema.push({name: 'updateTime', label: 'updated', dataType: 'STRING', semantics: this.TIMESTAMP_SEMANTICS});
+  
+  return {schema: schema};
+}
+
+
+/** 
+ * Predefined Data Studio function for fetching user-defined data. 
+ *
+ * @param {Object} request Data request parameters.
+ * @return {Object} Contains the schema and data for the given request.
+ */
+Connector.prototype.getData = function(request) {
+  const project = request.configParams.project;
+  if (!project) {
+    throw 'Missing project ID'; 
+  }
+  
+  const collection = request.configParams.collection;
+  if (!collection) {
+    throw 'Missing collection name'; 
+  }
+  
+  var numResultsValue = request.configParams.numResults;
+  if (!numResultsValue) {
+     numResultsValue = '1000';
+  }
+  const numResults = parseInt(numResultsValue);
```

Comment:
Please add a radix to this function. `parseInt(numResultsValue, 10)`

Reply:
Done